File: algorithms/05_csharp/math/gcd.cs
namespace Algorithms.Maths
{
    public static class NumberTheory
    {
        public static long Gcd(long a, long b)
        {
            while (b != 0)
            {
                long temp = b;
                b = a % b;
                a = temp;
            }
            return a;
        }

        public static long Lcm(long a, long b) => (a / Gcd(a, b)) * b;
    }
}

File: algorithms/08_r/math/gcd.R
gcd <- function(a, b) {
  while (b != 0) {
    temp <- b
    b <- a %% b
    a <- temp
  }
  return(a)
}

lcm <- function(a, b) {
  return(abs(a * b) / gcd(a, b))
}
